### PR TITLE
tests: arm_interrupt: remove redundant test case part

### LINKED ISSUE #34917: arch.interrupt.arm| arch.interrupt.extra_exception_info: lpcxpresso55s28 series: test failure
**Describe the bug**
interrupt testing failure, but is disable optimization and enlarge the stack size test can pass 

**To Reproduce**
1. west build -b lpcxpresso55s28 -- -DCONFIG_EXTRA_EXCEPTION_INFO=y
   or
    west build -b lpcxpresso55s28
3. make flash

test pass with below build options

west build -b lpcxpresso55s28 --  -DCONFIG_NO_OPTIMIZATIONS=y -DCONFIG_IDLE_STACK_SIZE=512 -DCONFIG_MAIN_STACK_SIZE=1024

**Expected behavior**
Caught system error -- reason 4

ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

	ESF could not be retrieved successfully. Shall never occur.

ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

	ESF could not be retrieved successfully. Shall never occur.

**Impact**
interrupt in lpc55 series

**Logs and console output**

```

*** Booting Zephyr OS build zephyr-v2.5.0-3077-g3a7e0f875cf1  ***

Running test suite arm_interrupt

===================================================================

START - test_arm_null_pointer_exception

Skipped

 PASS - test_arm_null_pointer_exception in 0.1 seconds

===================================================================

START - test_arm_interrupt

Available IRQ line: 59

E: >>> ZEPHYR FATAL ERROR 1: Unhandled interrupt on CPU 0

E: Current thread: 0x30000138 (unknown)

Caught system error -- reason 1

E: r0/a1:  0x00000003  r1/a2:  0x30002de0  r2/a3:  0x00000003

E: r3/a4:  0x30002bc0 r12/ip:  0xa0000000 r14/lr:  0x100025e7

E:  xpsr:  0x6100004b

E: EXC_RETURN: 0x0

E: Faulting instruction address (r15/pc): 0x1000070a

E: >>> ZEPHYR FATAL ERROR 3: Kernel oops on CPU 0

E: Fault during interrupt handling



E: Current thread: 0x30000138 (unknown)

Caught system error -- reason 3

E: r0/a1:  0x00000004  r1/a2:  0x30002de0  r2/a3:  0x00000004

E: r3/a4:  0x30002bc0 r12/ip:  0x00000000 r14/lr:  0x100025e7

E:  xpsr:  0x6100004b

E: EXC_RETURN: 0x0

E: Faulting instruction address (r15/pc): 0x10000728

E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0

E: Fault during interrupt handling



E: Current thread: 0x30000138 (unknown)

Caught system error -- reason 4

ASSERTION FAIL [0] @ WEST_TOPDIR/zephyr/tests/arch/arm/arm_interrupt/src/arm_interrupt.c:216

	Intentional assert



E: r0/a1:  0x00000004  r1/a2:  0x000000d8  r2/a3:  0x80000000

E: r3/a4:  0x0000004b r12/ip:  0xa0000000 r14/lr:  0x100025e7

E:  xpsr:  0x4100004b

E: EXC_RETURN: 0x0

E: Faulting instruction address (r15/pc): 0x1000b7b0

E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0

E: Fault during interrupt handling


E: Current thread: 0x30000138 (unknown)

Caught system error -- reason 4

ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

	ESF could not be retrieved successfully. Shall never occur.

ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

	ESF could not be retrieved successfully. Shall never occur.
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: zephyr-v2.5.0-3077-g3a7e0f875cf1

### LINKED ISSUE #35656: arch.interrupt.arm.arm_interrupt: hangs on mimxrt685_evk_cm33
**Describe the bug**
test application test_arm_interrupt hangs with below
Caught system error -- reason 4
Wrong crash type got 4 expected 0

**To Reproduce**
1.cd tests/arch/arm/arm_interrupt;  mkdir build; cd build
2. cmake -DBOARD=mimxrt685_evk_cm33
3. make; make flash

all test  configuration fails

  arch.interrupt.arm:
    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE
  arch.interrupt.no_optimizations:
    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE
    extra_configs:
      - CONFIG_NO_OPTIMIZATIONS=y
      - CONFIG_IDLE_STACK_SIZE=512
      - CONFIG_MAIN_STACK_SIZE=1024
  arch.interrupt.extra_exception_info:
    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE
    extra_configs:
      - CONFIG_EXTRA_EXCEPTION_INFO=y

**Expected behavior**
test pass

**Impact**
interrupt handling

**Logs and console output**

```
*** Booting Zephyr OS build v2.6.0-rc1-323-gae4fb20f6733  ***
Running test suite arm_interrupt
===================================================================
START - test_arm_null_pointer_exception
Skipped
PASS - test_arm_null_pointer_exception in 0.1 seconds
===================================================================
START - test_arm_interrupt
Available IRQ line: 59
E: &gt;&gt;&gt; ZEPHYR FATAL ERROR 1: Unhandled interrupt on CPU 0
E: Current thread: 0x30180140 (test_arm_interrupt)
Caught system error -- reason 1
E: r0/a1:  0x00000003  r1/a2:  0x301831d8  r2/a3:  0x00000003
E: r3/a4:  0x30182fc0 r12/ip:  0xa0000000 r14/lr:  0x18003f0f
E:  xpsr:  0x6100004b
E: Faulting instruction address (r15/pc): 0x1800190a
E: &gt;&gt;&gt; ZEPHYR FATAL ERROR 3: Kernel oops on CPU 0
E: Fault during interrupt handling
E: Current thread: 0x30180140 (test_arm_interrupt)
Caught system error -- reason 3
E: r0/a1:  0x00000004  r1/a2:  0x301831d8  r2/a3:  0x00000004
E: r3/a4:  0x30182fc0 r12/ip:  0x00000000 r14/lr:  0x18003f0f
E:  xpsr:  0x6100004b
E: Faulting instruction address (r15/pc): 0x18001928
E: &gt;&gt;&gt; ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Fault during interrupt handling
E: Current thread: 0x30180140 (test_arm_interrupt)
Caught system error -- reason 4
ASSERTION FAIL [0] @ WEST_TOPDIR/zephyr/tests/arch/arm/arm_interrupt/src/arm_interrupt.c:216
Intentional assert
E: r0/a1:  0x00000004  r1/a2:  0x000000d8  r2/a3:  0x80000000
E: r3/a4:  0x0000004b r12/ip:  0xa0000000 r14/lr:  0x18003f0f
E:  xpsr:  0x4100004b
E: Faulting instruction address (r15/pc): 0x1800d0a0
E: &gt;&gt;&gt; ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Fault during interrupt handling
E: Current thread: 0x30180140 (test_arm_interrupt)
Caught system error -- reason 4
ASSERTION FAIL [0] @ WEST_TOPDIR/zephyr/drivers/timer/sys_clock_init.c:23
E: r0/a1:  0x00000004  r1/a2:  0x00000017  r2/a3:  0x80000000
E: r3/a4:  0x0000000f r12/ip:  0xa0000000 r14/lr:  0xfffffffd
E:  xpsr:  0x4100000f
E: Faulting instruction address (r15/pc): 0x1800d0a0
E: &gt;&gt;&gt; ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Fault during interrupt handling
E: Current thread: 0x30180140 (test_arm_interrupt)
Caught system error -- reason 4
Wrong crash type got 4 expected 0
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux )
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v2.6.0-rc1-158-gcd9421a45e72

### PR DESCRIPTION
**Pull request by ioannisg**
The test case part in test_arm_interrupt, which is
triggering a SysTick IRQ (when SysTick is not used
for system timer implementation), to catch a spurious
interrupt is not really testing what it is designed
for. The reason is that the z_arm_exc_spurious has
be re-designed to be invoked only if the kernel is
built without support for system clock, i.e. with
CONFIG_SYS_CLOCK_EXISTS=n, but this configuration is
not compatible with ZTESTs and is, therefore, never
tested in this test, in any configuration. When the
SysTick is not implemented in the HW, it is not
possible to set the PendSTSET bit, so this is not
possible to execute this part of the test case, anyways.

What we have been catching as an error, here, is the
ASSERT in the sys_clock_isr() weak implementation, but
asserts are verified earlier in this test case, so this
part is really redundant.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Fixes #34917
Fixes #35656

### CODE REVIEW

**Comment by ioannisg** [comment]
@dleach02 this is supposed to fix the reported error in #34917

**Review by hakehuang — APPROVED** [review]

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c [hunk 1]
```diff
@@ -215,15 +215,6 @@ void arm_isr_handler(const void *args)
...
 	} else if (test_flag == 4) {
-#if defined(CONFIG_CPU_CORTEX_M_HAS_SYSTICK)
-#if !defined(CONFIG_SYS_CLOCK_EXISTS) || !defined(CONFIG_CORTEX_M_SYSTICK)
-		expected_reason = K_ERR_CPU_EXCEPTION;
-		SCB->ICSR |= SCB_ICSR_PENDSTSET_Msk;
-		__DSB();
-		__ISB();
-#endif
-#endif
-	} else if (test_flag == 5) {
```
Comment (stephanosio) [inline tests/arch/arm/arm_interrupt/src/arm_interrupt.c:226]: https://github.com/zephyrproject-rtos/zephyr/blob/92e370440e7c97255971678e3af7701ce4e6cc59/tests/arch/arm/arm_interrupt/src/arm_interrupt.c#L203

Noting that `arm_isr_handler` does `test_flag++` prior to checking it, the following will not be properly validated:

https://github.com/zephyrproject-rtos/zephyr/blob/92e370440e7c97255971678e3af7701ce4e6cc59/tests/arch/arm/arm_interrupt/src/arm_interrupt.c#L342

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c (hunk 1 above)
Comment (ioannisg) [reply tests/arch/arm/arm_interrupt/src/arm_interrupt.c:226]: Thanks, I had forgotten to remove this. It was added as part of the test-case that I am removing here.
Removed this now.

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c [hunk 2]
```diff
@@ -370,9 +338,6 @@ void test_arm_interrupt(void)
 	NVIC_EnableIRQ(i);
 	NVIC_SetPendingIRQ(i);
 
-	/* Set test flag so the IRQ handler executes the appropriate case. */
-	test_flag = 4;
```
Comment (ioannisg) [inline tests/arch/arm/arm_interrupt/src/arm_interrupt.c:374]: Removed here, @stephanosio

**Review by stephanosio — APPROVED** [review]

**Pull request merged**